contracts/v2/sovereignChains/AggOracleCommittee.sol

```diff
@@ -0,0 +1,344 @@
+
+    /**
+     * @notice Consolidate a global exit root that has reached quorum.
+     * This function it's meant to be called if the quorum was lowered, and there's a GER that has
+     * enough votes to be consolidated after updating it. Otherwise the consolidation happens automatically.
+     * @param globalExitRoot Global exit root to consolidate
+     */
+    function consolidateGlobalExitRoot(bytes32 globalExitRoot) external {
+        Report memory currentVotedReport = proposedGERToReport[globalExitRoot];
+
+        // Global exit root must have reached the quorum
+        // Notice that quorum cannot be 0
+        require(currentVotedReport.votes >= quorum, QuorumNotReached());
+
+        _consolidateGlobalExitRoot(globalExitRoot);
+    }
+
+    /**
+     * @notice Internal function to consolidate a global exit root.
+     * @param globalExitRoot Global exit root to consolidate
+     */
+    function _consolidateGlobalExitRoot(bytes32 globalExitRoot) internal {
+        // Delete the report
+        delete proposedGERToReport[globalExitRoot];
+
+        // Consolidate report
+        globalExitRootManagerL2Sovereign.insertGlobalExitRoot(globalExitRoot);
+        emit ConsolidatedGlobalExitRoot(globalExitRoot);
+    }
+
+    ////////////////////////
+    // Owner functions
+    ////////////////////////
+
+    /**
+     * @notice Add an oracle member.
+     * Only the owner can call this function.
+     * @param newOracleMember Address of the new oracle member
+     */
+    function addOracleMember(address newOracleMember) external onlyOwner {
+        _addOracleMember(newOracleMember);
+    }
+
+    /**
+     * @notice Internal function to add an oracle member.
+     * @param newOracleMember Address of the new oracle member
+     */
+    function _addOracleMember(address newOracleMember) internal {
+        require(newOracleMember != address(0), OracleMemberCannotBeZero());
+
+        require(
+            addressToLastProposedGER[newOracleMember] == bytes32(0),
+            AlreadyOracleMember()
+        );
+
+        // Add oracle member
+        addressToLastProposedGER[newOracleMember] = INITIAL_PROPOSED_GER;
+
+        // Add oracle member to the oracleMembers array
+        aggOracleMembers.push(newOracleMember);
+
+        emit AddAggOracleMember(newOracleMember);
+    }
+
+    /**
+     * @notice Remove an oracle member.
+     * Only the owner can call this function.
+     * @param oracleMemberAddress Address of the oracle member to remove
+     * @param oracleMemberIndex Index of the oracle member to remove
+     */
+    function removeOracleMember(
+        address oracleMemberAddress,
+        uint256 oracleMemberIndex
+    ) external onlyOwner {
+        bytes32 lastVotedReportHash = addressToLastProposedGER[
+            oracleMemberAddress
+        ];
+
+        require(lastVotedReportHash != bytes32(0), WasNotOracleMember());
+
+        require(
+            aggOracleMembers[oracleMemberIndex] == oracleMemberAddress,
```

Comment:
if `oracleMemberIndex` is greater or equal than the array length, solidity will trigger a revert and will not hit return teh revert reason `OracleMemberIndexMismatch()`.
Addition check should be done in order to return a meaningful revert reason if `oracleMemberIndex  >= aggOracleMembers.length`

Reply:
oki will fix